metricbeat/module/mssql/io/data.go

```diff
@@ -0,0 +1,80 @@
+package io
+
+import (
+	s "github.com/elastic/beats/libbeat/common/schema"
+	c "github.com/elastic/beats/libbeat/common/schema/mapstrstr"
+	"time"
+)
+
+var (
+	schema = s.Schema{
+		//Returns the list of tape devices and the status of mount requests for backups.
+		"backup_tapes": s.Object{
+			"physical_device_name":    c.Str("physical_device_name"),
+			"logical_device_name":     c.Str("logical_device_name"),
+			"status":                  c.Int("status"),
+			"status_desc":             c.Str("status_desc"),
+			"mount_request_time":      c.Time(time.RFC3339, "mount_request_time"),
+			"mount_expiration_time":   c.Time(time.RFC3339, "mount_expiration_time"),
+			"database_name":           c.Str("database_name"),
+			"spid":                    c.Int("spid"),
+			"command":                 c.Int("command"),
+			"command_desc":            c.Str("command_desc"),
+			"media_family_id":         c.Int("media_family_id"),
+			"media_set_name":          c.Str("media_set_name"),
+			"media_set_guid":          c.Str("media_set_guid"),
+			"media_sequence_number":   c.Int("media_sequence_number"),
+			"tape_operation":          c.Int("tape_operation"),
+			"tape_operation_desc":     c.Str("tape_operation_desc"),
+			"mount_request_type":      c.Int("mount_request_type"),
+			"mount_request_type_desc": c.Str("mount_request_type_desc"),
+		},
+		//This view returns the drive name of each of the shared drives if the current server instance is a
+		// clustered server. If the current server instance is not a clustered instance it returns an empty rowset.
+		"cluster_shared_drives": s.Object{
+			"drive_name":  c.Str("DriveName"),
+			"pdw_node_id": c.Int("pdw_node_id"),
+		},
+		//Returns a row for each pending I/O request in SQL Server.
+		"pending_io_requests": s.Object{
+			"io_completion_request_address": c.Str("io_completion_request_address"),
+			"io_type":                       c.Str("io_type"),
+			"io_pending":                    c.Int("io_pending"),
+			"io_completion_routine_address": c.Str("io_completion_routine_address"),
+			"io_user_data_address":          c.Str("io_user_data_address"),
+			"scheduler_address":             c.Str("scheduler_address"),
+			"io_handle":                     c.Str("io_handle"),
+			"io_offset":                     c.Int("io_offset"),
+			"io_pending_ms_ticks":           c.Int("io_pending_ms_ticks"),
+			"pdw_node_id":                   c.Int("pdw_node_id"),
+		},
+		//Returns I/O statistics for data and log files. This dynamic management view replaces the
+		// fn_virtualfilestats function.
+		"virtual_file_stats": s.Object{
+			"database_name":            c.Str("database_name"),
+			"database_id":              c.Int("database_id"),
```

Comment:
You could structure these fields in an object:
```
"database": s.Object{
  "name": c.Str("database_name"),
  "id":   c.Int("database_id"),
}
```

Reply:
Because now there is only a single group of metrics in this package, I haven't done it to simplify the tree.

Reply:
Well, even with an only metricset I think this would be something good to do in any case.